### ISSUE TITLE
Inspect Solr query builder, some queries generate lots of Boolean expressions

### ISSUE THREAD

**Issue by flekschas** [refinery-platform/refinery-platform#952]
Currently running http://192.168.50.50:8983/solr/core/select/?allIds=1&defType=synonym_edismax&f.description.hl.alternateField=description&f.title.hl.alternateField=title&fl=dbid,uuid,access&fq=django_ct:core.dataset&hl=true&hl.fl=title,description&hl.maxAlternateFieldLength=128&hl.simple.post=%3C%2Fem%3E&hl.simple.pre=%3Cem%3E&q=mouse+cortical+neurospheres+in+response+to+Fezf2+over-expression&qf=title%5E0.5+accession+submitter+text&rows=50&start=0&synonyms=true&wt=json will lead to a `maxClauseCount is set to 1024` error.
The reason for that is that the query is somehow transformed into many (more than 1024) Boolean expressions.

Ideas:
- This might be because of the ngram field for the title.
- Synonym search

Anyway the cause should be investigated it might be resource inefficient.

tl/dr

For now I'll increase `maxClauseCount` is it's an arbitrary limit and shouldn't hit the performance.

Some stuff to read
- https://wiki.apache.org/lucene-java/LuceneFAQ#Why_am_I_getting_a_TooManyClauses_exception.3F
- http://stackoverflow.com/q/1534789/981933

**Comment by mccalluc** [comment]
Trying to clean up old issues.
- I have tried both port 8000 and 8983 locally, and although the results are empty, there is no error.
- When I look in the solr log, I don't see an explosion:
```
2017-07-18 20:02:59.866 INFO  (qtp945299648-18) [   x:core] o.a.s.c.S.Request [core] webapp=/solr path=/select params={allIds=1&f.description.hl.alternateField=description&fl=dbid,uuid,access&start=0&hl.maxAlternateFieldLength=128&q=mouse+cortical+neurospheres+in+response+to+Fezf2+over-expression&f.title.hl.alternateField=title&qf=title^0.5+accession+submitter+text&synonyms=true&hl.simple.pre=<em>&hl.simple.post=</em>&hl.fl=title,description&wt=json&hl=true&fq=django_ct:core.dataset+AND+access:(g_100)&rows=50&defType=synonym_edismax} hits=0 status=0 QTime=6 
2017-07-18 20:02:59.878 INFO  (qtp945299648-14) [   x:core] o.a.s.c.S.Request [core] webapp=/solr path=/select params={fl=dbid&start=0&q=mouse+cortical+neurospheres+in+response+to+Fezf2+over-expression&qf=title^0.5+accession+submitter+text&wt=json&fq=django_ct:core.dataset+AND+access:(g_100)&rows=2147483647&defType=synonym_edismax} hits=0 status=0 QTime=5 
```

@flekschas : I'm not sure that two requests are really necessary, but I don't see the query blow-up you describe. Can this be closed?

**Comment by flekschas** [comment]
@mccalluc Yes they are. The first one returns all ids, which are needed for initializing the visualizations. The params of the two queries are very different too.

How many Boolean queries do you count? (Just curious)

**Comment by mccalluc** [comment]
> How many Boolean queries do you count?

Each of the log lines I quoted has one `AND`, so two, total? I'm not sure that we're talking about the same thing. In the original post, you describe getting more than 1024 Boolean expressions, but either the behavior has changed since then, or I don't understand how to replicate the problem.

If there is still a problem here, what is the behavior we would like to see?

**Comment by flekschas** [comment]
Where exactly do you see the number of Boolean clauses? Maybe I am missing something but I only see a time stamp, some other stuff, the parameters, and hits, status, and qtime.

You won't get a `maxClauseCount ` error because because I increased it such that the error goes away.

You can close this issue if you like but it might pop up again. Whether one should care? I don't know.

**Comment by mccalluc** [comment]
Maybe what you mean by boolean clause is different than what I'm looking at?: I see
```
fq=django_ct:core.dataset+AND+access:(g_100)
```
in both of those log lines, but maybe you're talking about something else?

**Comment by flekschas** [comment]
Please check the two links I provided. The queries you look at are user-defined queries. Solr transforms those further. The issue I am talking about is the final query that is actually executed by Solr in the end.

**Comment by jkmarx** [comment]
Ref #2950
